remote.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/keybase/go-logging"
+	"github.com/keybase/go-updater/util"
+)
+
+// RemoteUpdateSource finds releases/updates from custom url feed (used primarily for testing)
+type RemoteUpdateSource struct {
+	defaultURI string
+	log        logging.Logger
+}
+
+// NewRemoteUpdateSource builds remote update source without defaults. The url used is passed
+// via options instead.
+func NewRemoteUpdateSource(defaultURI string, log logging.Logger) RemoteUpdateSource {
+	return RemoteUpdateSource{
+		defaultURI: defaultURI,
+		log:        log,
+	}
+}
+
+// Description returns update source description
+func (r RemoteUpdateSource) Description() string {
+	return "Remote"
+}
+
+func (r RemoteUpdateSource) sourceURL(options UpdateOptions) string {
+	params := util.JoinPredicate([]string{options.Platform, options.Env, options.Channel}, "-", func(s string) bool { return s != "" })
```

Comment:
can't this result in an ambiguous URI? `[]string{"foo", "", "foo"}` will resolve to the same URI as `[]string{"", "foo", "foo"}`.